capella_model_explorer/backend/main.py

```diff
@@ -0,0 +1,332 @@
+            model=state.model,
+            diff_data=state.diff,
+            object_diff=state.object_diff,
+        )
+        return fastapi.responses.HTMLResponse(
+            content=rendered, status_code=200
+        )
+    except jinja2.TemplateSyntaxError as e:
+        error_message = markupsafe.Markup(
+            "<p style='color:red'>Template syntax error: {}, line {}</p>"
+        ).format(e.message, e.lineno)
+        base.error = error_message
+        print(base)
+        return fastapi.responses.HTMLResponse(content=error_message)
+    except Exception as e:
+        logger.exception("Error rendering template")
+        trace = markupsafe.escape(traceback.format_exc())
+        error_message = markupsafe.Markup(
+            '<p style="color:red">'
+            f"Unexpected error: {type(e).__name__}: {e}"
+            '</p><pre style="font-size:80%;overflow:scroll">'
+            f"object={object!r}\nmodel={state.model!r}"
+            f"\n\n{trace}"
+            "</pre>"
+        )
+        return fastapi.responses.HTMLResponse(content=error_message)
+
+
+app = fastapi.FastAPI(version=capella_model_explorer.__version__)
+app.add_middleware(
+    fastapi.middleware.cors.CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+router = fastapi.APIRouter(prefix=ROUTE_PREFIX)
+app.include_router(router)
+app.mount(
+    f"{ROUTE_PREFIX}/assets",
+    fastapi.staticfiles.StaticFiles(directory=FRONTEND_DIR / "assets"),
+)
+app.mount(
+    f"{ROUTE_PREFIX}/static",
+    fastapi.staticfiles.StaticFiles(directory=FRONTEND_DIR / "static"),
+)
+
+try:
+    MODEL_INFO = os.environ["MODEL"]
+except KeyError as err:
+    raise SystemExit("MODEL environment variable is not set") from err
+
+try:
+    _ = state.model
+except AttributeError:
+    state.model = capellambse.loadcli(MODEL_INFO)
+
+state.templates = fastapi.templating.Jinja2Templates(directory=FRONTEND_DIR)
+
+state.templates_path = pathlib.Path(
+    os.getenv("TEMPLATES_DIR", str(TEMPLATES_DIR))
+)
+state.templates_loader = tl.TemplateLoader(state.model)
+state.jinja_env = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(state.templates_path)
+)
+state.jinja_env.finalize = _finalize
+state.jinja_env.filters["make_href"] = _make_href_filter
+state.templates_index = state.templates_loader.index_path(state.templates_path)
+
+
+@app.middleware("http")
+async def update_last_interaction_time(request: fastapi.Request, call_next):
+    if request.url.path not in ("/metrics", "/favicon.ico"):
+        state.last_interaction = time.time()
+    return await call_next(request)
+
+
+@app.get("/metrics")
+def metrics():
+    idle_time_minutes = (time.time() - state.last_interaction) / 60
+    state.idle_time_gauge.set(idle_time_minutes)
+    return fastapi.Response(
+        content=prometheus_client.generate_latest(),
+        media_type="text/plain",
+    )
+
+
+@app.router.get("/api/commits")
```

Comment:
We've declared a router further up that includes the `ROUTE_PREFIX`, but here we're using the `app.router`, which is the main router without that prefix.

```diff
-@app.router.get("/api/commits")
+@router.get("/api/commits")
```

Same applies to all other routes decorated with `@app.router.xyz`.

Reply:
Thanks for this important observation. Is fixed.